src/vm/script.rs

```diff
@@ -322,6 +322,129 @@ impl Script {
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomUnsigned64Var) => {
+                            frame.stack.push(VmTerm::Unsigned64(rng.gen::<u64>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 8;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomUnsigned128Var) => {
+                            frame.stack.push(VmTerm::Unsigned128(rng.gen::<u128>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 16;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomSigned8Var) => {
+                            frame.stack.push(VmTerm::Signed8(rng.gen::<i8>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 1;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomSigned16Var) => {
+                            frame.stack.push(VmTerm::Signed16(rng.gen::<i16>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 2;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomSigned32Var) => {
+                            frame.stack.push(VmTerm::Signed32(rng.gen::<i32>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 4;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomSigned64Var) => {
+                            frame.stack.push(VmTerm::Signed64(rng.gen::<i64>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 8;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomSigned128Var) => {
+                            frame.stack.push(VmTerm::Signed128(rng.gen::<i128>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 16;
+                        }
+
+                        ScriptExecutorState::ExpectingBytesOrCachedTerm(OP::Hash160Var) => {
+                            let mut arr: [u8; 20] = [0; 20];
+
+                            // TODO
```

Comment:
Here, just push the empty buffer on the execution stack and then use `mem::copy` for each byte